CMakeLists.txt

```diff
@@ -145,8 +145,8 @@ if (NOT Boost_FOUND)
   endif()
 
   # build dir needs to be added, that's where conan writes some info
-  list(APPEND CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR})
-  list(APPEND CMAKE_PREFIX_PATH ${CMAKE_BINARY_DIR})
+  list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_BINARY_DIR})
+  list(APPEND CMAKE_PREFIX_PATH ${CMAKE_CURRENT_BINARY_DIR})
```

Comment:
@nilsnolde This is the major thing that prevents build as submodule, because of this:
- building standalone Valhalla, it is a top-level project and `CMAKE_BINARY_DIR == CMAKE_CURRENT_BINARY_DIR`
- building Valhalla a subdirectory, then `CMAKE_BINARY_DIR != CMAKE_CURRENT_BINARY_DIR` which breaks the Valhalla's expected `CMAKE_MODULE_PATH`

Reply:
Oh yeah I get that. Was more referring to the linked issue, where @yzsolt wasnt able to include Valhalla without the mjolnir/proto workaround in this PR. Not that it matters greatly, mjolnir is hardly a module one would like to have in a superproject 😄

Reply:
Yeah I've done this before but only for the purposes of android and ios builds in which I disable data tools, which is all of mjolnir